anomalib/models/patchcore/torch_model.py

```diff
@@ -178,15 +180,15 @@ def compute_anomaly_score(self, patch_scores: Tensor, locations: Tensor, embeddi
         # 1. Find the patch with the largest distance to it's nearest neighbor in each image
         max_patches = torch.argmax(patch_scores, dim=1)  # (m^test,* in the paper)
         # 2. Find the distance of the patch to it's nearest neighbor, and the location of the nn in the membank
-        score = patch_scores[torch.arange(len(patch_scores)), max_patches]  # s in the paper
-        nn_index = locations[torch.arange(len(patch_scores)), max_patches]  # m^* in the paper
+        score = patch_scores[torch.arange(patch_scores.shape[0]), max_patches]  # s in the paper
```

Comment:
for a tensor `x`, `len(x)` should return the size of the first dimension of `x`. So I don't think this is needed either.

Reply:
it may be a bit more robust though, so we could keep it :)

Reply:
Yeah, torch suggested this usage instead of `len`, which is the reason for this change.